### ISSUE TITLE
[Bug]: Cannot execute chrome for testing in read only docker "--database is required"

### ISSUE THREAD

**Issue by furstenheim-goodnotes** [puppeteer/puppeteer#11023]
### Steps to reproduce

Dockerfile:

```
FROM node:18-slim
RUN npx @puppeteer/browsers install chrome@116.0.5793.0 --path /usr/bin/chrome-install

RUN apt-get update \
    && apt-get install -y wget gnupg dumb-init \
          ## chromium dependencies https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#chrome-doesnt-launch-on-linux
          ca-certificates \
          fonts-liberation \
          libasound2 \
          libatk-bridge2.0-0 \
          libatk1.0-0 \
          libc6 \
          libcairo2 \
          libcups2 \
          libdbus-1-3 \
          libexpat1 \
          libfontconfig1 \
          libgbm1 \
          libgcc1 \
          libglib2.0-0 \
          libgtk-3-0 \
          libnspr4 \
          libnss3 \
          libpango-1.0-0 \
          libpangocairo-1.0-0 \
          libstdc++6 \
          libx11-6 \
          libx11-xcb1 \
          libxcb1 \
          libxcomposite1 \
          libxcursor1 \
          libxdamage1 \
          libxext6 \
          libxfixes3 \
          libxi6 \
          libxrandr2 \
          libxrender1 \
          libxss1 \
          libxtst6 \
          lsb-release \
          wget \
          xdg-utils \
      && apt-get update

ENTRYPOINT [ "/bin/bash", "-c" ]

CMD ["bash"]

```

In the terminal:

```
$ docker build -t test-puppeteer -f Dockerfile . && docker run --read-only -it test-puppeteer   bash
## inside docker
$ /usr/bin/chrome-install/chrome/linux-116.0.5793.0/chrome-linux64/chrome

```

### Expected results

I should be able to run the browser. Before chrome testing, I was using chromium and it was possible.

### Actual results

I get the following error
```
chrome_crashpad_handler: --database is required
Try 'chrome_crashpad_handler --help' for more information.
Trace/breakpoint trap (core dumped)
root@27814e039735:/# 
exit
```

**Comment by OrKoN** [comment]
@furstenheim-goodnotes could you try a non-slim image? I assume the slim one is based on the Alpine image?

**Comment by OrKoN** [comment]
@thiagowfx is it something for the CfT?

**Comment by furstenheim-goodnotes** [comment]
@OrKoN sure, which one do you want me to try?

**Comment by OrKoN** [comment]
@furstenheim-goodnotes the `node` one should work https://github.com/puppeteer/puppeteer/blob/main/docker/Dockerfile

**Comment by furstenheim-goodnotes** [comment]
`node` doesn't work either. Note, that the key here is the "read-only". If I remove `read-only` flag when running it works

**Comment by furstenheim-goodnotes** [comment]
Is there a way to run this in a read only environment? Chromium was running in such a way before (installing it with apt-get update...). It's better from a security perspective.

**Comment by OrKoN** [comment]
@furstenheim-goodnotes not sure, there should be no differences between Chrome and Chrome for Testing that would affect the behaviour here. Perhaps, the apt-get version creates a user data dir at the build time? Also, the --headless mode might run.

**Comment by furstenheim-goodnotes** [comment]
Using `--user-data-dir` on a folder that has permissions is not enough

```
mkdir tmp
## -v tmp:/tmp makes the volume writeable
docker build -t test-puppeteer -f Dockerfile . && docker run --read-only -v tmp:/tmp -it test-puppeteer   bash
### inside docker
### succeeds
$ touch /tmp/a
### succeeds
$ mkdir /tmp/profile 
> /usr/bin/chrome-folder/chrome --user-data-dir=/tmp/profile 
chrome_crashpad_handler: --database is required
Try 'chrome_crashpad_handler --help' for more information.
[191:191:0925/134323.066842:ERROR:socket.cc(120)] recvmsg: Connection reset by peer (104)
Trace/breakpoint trap
```

**Comment by OrKoN** [comment]
Does adding `~/.config` folder help?

**Comment by furstenheim-goodnotes** [comment]
Thanks @OrKoN, making .config also readable works.

**Comment by thiagowfx** [comment]
https://chromium.googlesource.com/chromium/src/+/457b3daf6fdbb82bbb3e29f4da806b1d48c723b1 could be related:

> Use ~/.config/google-chrome-for-testing instead of ~/.config/chromium.

**Comment by sfc-gh-selee** [comment]
We got this to work by setting in the environment:
```
XDG_CONFIG_HOME=/tmp/.chromium
XDG_CACHE_HOME=/tmp/.chromium
```

**Comment by galenhuntington** [comment]
I encountered a problem similar to this, and the culprit turned out to be this bug:
https://bugs.launchpad.net/ubuntu/+source/docker.io-app/+bug/2029523

Upgrading to a fixed version of Docker made it work again.

**Comment by Bloodsucker** [comment]
If anyone is coming to this issue because they encounter a Puppeteer/Chrome issue with the --database flag when running CI/CD on a Jenkins Pipeline + Docker on a linux host, then the solution from @sfc-gh-selee works. In my case:

```
XDG_CONFIG_HOME="$WORKSPACE@tmp/.chromium"
XDG_CACHE_HOME="$WORKSPACE@tmp/.chromium"
```

**Comment by agrinko** [comment]
@sfc-gh-selee thank you so much! 👍 
I've spent several days fighting with a similar issue. 
In my case, a docker container with Puppeteer worked fine locally (when running with Docker Desktop), but failed when deployed on an actual environment via Kubernetes. I couldn't figure out the root cause until I tried these magic variables. I'm still not sure what they do but I'll always keep them with me, just in case 😅
